Readability Archive

bcomnes opened the pull request:
WIP readability archive. 

- [x] API endpoints (rough draft)
- [x] Entity web pages
- [ ] Store plain text representation (comes out of the extraction, will be useful for full text search)
- [x] Article mode on archive view page
- [x] Archives listing
- [ ] Show the latest archive on the bookmarks permalink?
- [ ] Clean up static admin route rules
- [ ] Restrict loosened helmet rules to just the pages that need them.

bcomnes committed:
Add a button to create an archive

bcomnes committed:
Update bookmarks query to include archives

A new option to the query is called fullArchives which triggers the inclusion of full archive content. Not sure if this will be used yet or not.

bcomnes committed:
Fill out remaining archive routes

bcomnes committed:
Add a badge

bcomnes committed:
Checking archive pages

This is just mostly copy and paste between the episodes pages. I can de-dupe them at somepoint.

voxpelli commented:
If you want to try out other libraries, here's what I found:

* https://github.com/postlight/parser
* https://github.com/extractus/article-extractor
* https://github.com/ageitgey/node-unfluff (probably abandoned though)

bcomnes commented:
The postlight one seems like the most comprehensive. Plan is to try it out if the Mozilla one doesn't work well next. It has the most site specific extractor code out of all of them. I can't tell if this is a required dedication to detail or a deficiency in their general approach.

bcomnes committed:
Working

bcomnes commented:
<img width="1278" alt="Screenshot 2023-03-26 at 4 40 24 PM" src="https://user-images.githubusercontent.com/166301/227809282-7ae351d0-70f9-4838-af55-7edcbbacb220.png">

Everything piped together. Now I just need to clean it up and make it presentable.

bcomnes committed:
Finish readability archive and add bookmark summary field

The summary field is used mostly for client extracted metadata from the page, separate from the readability archive step. Its user editable, which distinguishes it from the server extracted excerpt.
